apis/management.cattle.io/v3/machine_types.go

```diff
@@ -193,8 +196,10 @@ type NodeSpec struct {
 	DisplayName              string            `json:"displayName"`
 	RequestedHostname        string            `json:"requestedHostname,omitempty" norman:"type=hostname,nullable,noupdate,required"`
 	InternalNodeSpec         v1.NodeSpec       `json:"internalNodeSpec"`
+	DesiredNodeTaints        []v1.Taint        `json:"desiredNodeTaints"`
 	DesiredNodeLabels        map[string]string `json:"desiredNodeLabels,omitempty"`
 	DesiredNodeAnnotations   map[string]string `json:"desiredNodeAnnotations,omitempty"`
+	UpdateTaintsFromAPI      *bool             `json:"updateTaintsFromAPI,omitempty" norman:"default=false"`
```

Comment:
Can you describe how this field is used, and why it needs to be a pointer when it defaults to false

Reply:
The default value is not good. Will be removed.
This field will be set to true by the store in Rancher. It means there are some changes from the api. 
And it will be set to false by the controller. It means there are some changes from the controller.
The linux taints controller won't update the `v3.Node` object if the `UpdateTaintsFromAPI` is not nil. It can avoid multiple changes in the same time and the second change overwrite the first change.